A32NX/html_ui/Pages/A32NX_Core/FMGC/A32NX_FlightPhaseManager.js

```diff
@@ -0,0 +1,307 @@
+ *                      gets called to directly verify the current flight phase is still valid.
+ *
+ *
+ * A32NX_FlightPhase Object:
+ *
+ * Variables:
+ *  nextFmgcFlightPhase: This variable defines the next flight phase the FMGC can automatically transition into.
+ *                       If not defined, the FMGC won't be able to transition onto another flight phase automatically
+ *                       without pilot input, if that's the case the check method will always return false.
+ *                       This variable can be found in the constructor or init of the flight phase object.
+ *                       Whether it's defined in the constructor or init depends on whether the variable can be changed
+ *                       and hence the default transition flight phase be manipulated, if that's the case the variable
+ *                       will be defined in init method to ensure the variable is always in it's intended state.
+ *
+ * Methods:
+ *  init:               This method is called once every time the FMGC transitions into this flight phase to initialize
+ *                      the current flight phase variables if needed.
+ *                      (This method gets passed the fmc object to have access to the fmc variables and methods)
+ *
+ *  check:              This method returns as boolean.
+ *                      True: Transition into next flight phase will be initiated
+ *                      (The next flight phase is defined as nextFmgcFlightPhase)
+ *                      False: The current FMGC flight phase maintains is maintained.
+ *                      (This method gets passed the fmc object to have access to the fmc variables and methods)
+ */
+class A32NX_FlightPhaseManager {
+    constructor(_fmc) {
+        console.log('A32NX_FlightPhaseManager constructed');
+        this.fmc = _fmc;
+
+        this.flightPhases = {
+            [FmgcFlightPhases.PREFLIGHT]: new A32NX_FlightPhase_PreFlight(),
+            [FmgcFlightPhases.TAKEOFF]: new A32NX_FlightPhase_TakeOff(),
+            [FmgcFlightPhases.CLIMB]: new A32NX_FlightPhase_Climb(),
+            [FmgcFlightPhases.CRUISE]: new A32NX_FlightPhase_Cruise(),
+            [FmgcFlightPhases.DESCENT]: new A32NX_FlightPhase_Descent(),
+            [FmgcFlightPhases.APPROACH]: new A32NX_FlightPhase_Approach(),
+            [FmgcFlightPhases.GOAROUND]: new A32NX_FlightPhase_GoAround(),
+            [FmgcFlightPhases.DONE]: new A32NX_FlightPhase_Done()
+        };
+
+        this.activeFlightPhase = this.flightPhases[FmgcFlightPhases.PREFLIGHT];
+
+        SimVar.SetSimVarValue("L:A32NX_FMGC_FLIGHT_PHASE", "number", FmgcFlightPhases.PREFLIGHT);
+
+        this.activeFlightPhase.init(_fmc);
+    }
+
+    checkFlightPhase() {
+        if (this.activeFlightPhase.check(this.fmc)) {
+            this.changeFlightPhase(this.activeFlightPhase.nextFmgcFlightPhase);
+        }
+    }
+
+    handleFcuInput() {
```

Comment:
Flight Phases only react to alt knob push/pull, so I find that naming perfectly fine.

Reply:
I'm fine with either way so leave as is and mark this resolved if you like. I try to look at APIs on their own merit rather than from the lens of a specific consumer, even if it seems obvious now that there wouldn't be any new consumers later.